hs_geographic_feature_resource/receivers.py

```diff
@@ -0,0 +1,522 @@
+        for file in files:
+            if file[len(file)-1]=='/':
+                dir_count+=1
+            else:
+                fileName, fileExtension = os.path.splitext(file)
+                if ".shp" == fileExtension:
+                    shp_filename = fileName
+                    shp=True
+                elif ".shx" == fileExtension:
+                    shx_filename = fileName
+                    shx=True
+                elif ".dbf" == fileExtension:
+                    dbf_filename = fileName
+                    dbf=True
+
+        if shp_filename == shx_filename and shp_filename == dbf_filename:
+            all_have_same_filename = True
+
+    if shp & shx & dbf & all_have_same_filename & (dir_count<=1):
+        return True
+    else:
+        return False
+
+
+def is_zipped_shapefiles(files):
+#check if the uploaded zip files contains valid shapefiles (shp, shx, dbf)
+    if(len(files) == 1) and '.zip' in files[0].name:
+        zipfile_path=files[0].file.name
+        if zipfile.is_zipfile(zipfile_path):
+            zf = zipfile.ZipFile(zipfile_path, 'r')
+            content_fn_list = zf.namelist()
+            return check_fn_for_shp(content_fn_list)
+    return False
+
+
+#
+# def is_spatialite(files):
+# # check for sqlite
+#     for file in files:
+#         if ".sqlite" in file.name:
+#             return True
+#     return False
+
+
+def check_uploaded_files_type(files):
+        rslt={}
+        if is_shapefiles(files):
+            uploaded_file_type = "shp"
+            rslt["uploaded_file_type"]=uploaded_file_type
+            #create a temp foler and copy shapefiles
+            tmp_dir=tempfile.mkdtemp()
+            rslt["tmp_dir"]=tmp_dir
+            baseFilename=None
+            files_list_for_extract_metadata = []
+            for file in files:
+                source = file.file.name
+                fileName, fileExtension = os.path.splitext(file.name.lower())
+                if fileExtension == ".shp":
+                    baseFilename=fileName
+                #target = tmp_dir + "/" + res_titile_fn + fileExtension
+                target = tmp_dir + "/" + fileName + fileExtension
+                shutil.copy(source, target)
+
+            #shp_full_path = tmp_dir + "/" + res_titile_fn + ".shp"
+            shp_full_path = tmp_dir + "/" + fileName + ".shp"
+            rslt['shp_full_path']=shp_full_path
+
+            rslt["baseFilename"]=baseFilename
+            uploadedFileCount=len(files)
+            rslt["uploadedFileCount"]=uploadedFileCount
+            uploadedFilenameString_dict={}
+            for i in range(len(files)):
+                uploadedFilenameString_dict[files[i].name]=str(i)
+            uploadedFilenameString = json.dumps(uploadedFilenameString_dict)
+            rslt["uploadedFilenameString"]=uploadedFilenameString
+            rslt["are_files_valid"] = True
+            rslt['message'] = 'All files are validated.'
+
+        elif is_zipped_shapefiles(files):
```

Comment:
These two blocks are nearly entirely identical. If you unpack the zipfile first you could use the same processing logic on either case, without repeating it.